userspace/libsinsp/event.cpp

```diff
@@ -2747,3 +2747,81 @@ uint64_t sinsp_evt::get_lastevent_ts() const
 {
 	return m_tinfo->m_lastevent_ts;
 }
+
+bool sinsp_evt::clone_event(sinsp_evt &dest, const sinsp_evt &src)
+{
+	dest.m_inspector = src.m_inspector;
+
+	if (src.m_pevt != nullptr)
+	{
+		dest.m_pevt_storage = new char[src.m_pevt->len];
+		memcpy(dest.m_pevt_storage, src.m_pevt, src.m_pevt->len);
+		dest.m_pevt = (scap_evt *) dest.m_pevt_storage;
+	}
+	else
+	{
+		dest.m_pevt_storage = nullptr;
+		dest.m_pevt = nullptr;
+	}
+
+	dest.m_poriginal_evt = nullptr;
+
+	// tinfo
+	if (src.m_tinfo_ref && src.m_tinfo && src.m_tinfo_ref.get() != src.m_tinfo)
+	{
+		// bad data
+		return false;
+	}
+
+	dest.m_tinfo_ref = src.m_tinfo_ref;
+	dest.m_tinfo = src.m_tinfo;
+	if (src.m_tinfo_ref)
+	{
+		dest.m_tinfo_ref = src.m_tinfo_ref;
+		dest.m_tinfo = dest.m_tinfo_ref.get();
+	}
+	else if (src.m_tinfo)
+	{
+		dest.m_tinfo_ref = dest.m_inspector->get_thread_ref(src.m_tinfo->m_tid, false, false);
+		if (dest.m_tinfo_ref == nullptr)
+		{
+			// no tinfo
+			return false;
+		}
+		dest.m_tinfo = dest.m_tinfo_ref.get();
+	}
+
+	dest.m_tinfo_ref = src.m_tinfo_ref;
+	dest.m_tinfo = src.m_tinfo;
```

Comment:
Doesn't this overwrite the values set before in the if/else block?

Reply:
Was gonna comment on this too, I think all that's really missing in the if-elseif block is an else condition to stop copying if both `src.m_tinfo_ref` and `src.m_tinfo` are null.